### PR TITLE
Add LensKit package

### PR DESCRIPTION
**Pull request by mdekstrand**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/lenskit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - pytest-runner
+    - pip
+    - numba >=0.43
+    - pandas >=0.24
+    - numpy >=1.16
+    - scipy >=1.2
+    - pyarrow >=0.15
+    - cffi >=1.12.2
+    - binpickle >=0.3.2
+    - mkl-devel  # to build helper library
```
Comment (scopatz) [inline recipes/lenskit/meta.yaml:31]: Are most of these actually needed to compile?

Comment (mdekstrand) [reply recipes/lenskit/meta.yaml:31]: They are in the source package's `install_requires`, so the package installation will try to find them if they aren't already installed.

Comment (scopatz) [reply recipes/lenskit/meta.yaml:31]: host is different than install thoug

Comment (chrisburr) [reply recipes/lenskit/meta.yaml:31]: `install_requires` is equivalent to conda's `run` (`host` is `setup_requires`)

Comment (mdekstrand) [reply recipes/lenskit/meta.yaml:24]: Ok. Moved.

File: recipes/lenskit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 985b7b545dfbed7eb939014f938e685332a40614265c4553fa4bb6d63b43a46e
+
+build:
+  number: 0
+  script: "{{ PYTHON }} setup.py install"
```
Comment (chrisburr) [inline recipes/lenskit/meta.yaml:14]: https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

Comment (mdekstrand) [reply recipes/lenskit/meta.yaml]: I have made it use `pip`. There still is a manual invocation of `setup.py`, but just to build the helper library. It isn't auto-built during `pip install`. The helper library is useless outside of Conda-based installations (it's only for working with MKL), so I haven't put effort in the upstream package into debugging why pip doesn't build it.

**Comment by mdekstrand** [comment]
I'm temporarily building from Git, because I'm working on the upstream changes necessary to make sure the conda-forge package builds cleanly, and want to test that before tagging the release upstream.

Once I've tagged 0.11.0 upstream and pushed to PyPI, I will update this PR and make it ready for review again.

**Comment by mdekstrand** [comment]
Ok, this thing is all fixed and ready to review again :)

**Review by dopplershift — APPROVED** [review]

**Pull request merged**